packages/accordions/src/elements/stepper/components/Step.tsx

```diff
@@ -5,19 +5,34 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import React, { forwardRef, useRef, LiHTMLAttributes } from 'react';
+import React, { forwardRef, LiHTMLAttributes, useEffect, useState } from 'react';
 import { StyledStep, StyledLine } from '../../../styled';
 import { StepContext, useStepperContext } from '../../../utils';
 
+const useStep = () => {
+  const { currentIndexRef, ...context } = useStepperContext();
+  const [index, setIndex] = useState(currentIndexRef.current);
+
+  useEffect(() => {
+    setIndex(currentIndexRef.current);
+    currentIndexRef.current++;
+    const currentIndex = currentIndexRef;
+
+    return () => {
+      currentIndex.current--;
```

Comment:
Is this decrement effect cleanup necessary? The `currentIndexRef` is set to zero whenever the parent `<Stepper />` is rendered.

Reply:
Sometimes this is re-rendered when the parent component is not re-rendered. This is what StrictMode does. It'll render these child components twice without rendering `<Stepper />`

Reply:
👍  Thanks for the explanation